app/ante/authz_guard.go

```diff
@@ -1,9 +1,51 @@
+// Copyright (c) 2023-2024 Nibi, Inc.
 package ante
 
-// TODO: https://github.com/NibiruChain/nibiru/issues/1915
-// feat(ante): Add an authz guard to disable authz Ethereum txs and provide
-// additional security around the default functionality exposed by the module.
-//
-// Implemenetation Notes
-// UD-NOTE - IsAuthzMessage fn. Use authz import with module name
-// UD-NOTE - Define set of disabled txMsgs
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/cosmos/gogoproto/proto"
+
+	"github.com/NibiruChain/nibiru/v2/x/evm"
+)
+
+var genericAuthTypeTurl = "/" + proto.MessageName(&authz.GenericAuthorization{})
+
+// AnteDecoratorAuthzGuard filters autz messages
+type AnteDecoratorAuthzGuard struct{}
+
+// AnteHandle rejects "authz grant generic --msg-type '/eth.evm.v1.MsgEthereumTx'"
+func (rmd AnteDecoratorAuthzGuard) AnteHandle(
+	ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler,
+) (newCtx sdk.Context, err error) {
+	for _, msg := range tx.GetMsgs() {
+		if msgGrant, ok := msg.(*authz.MsgGrant); ok {
```

Comment:
`Revoke` seems harmless. I would also remove `Exec` as well to address [this vulnerability from Jump](https://jumpcrypto.com/writing/bypassing-ethermint-ante-handlers/).
- https://github.com/NibiruChain/nibiru/issues/2027